### DISCUSSION TITLE
Best practices around useContractWrite mode

### DISCUSSION THREAD

**Discussion by ItsMeTazz** [wevm/wagmi#1970] [Q&A]
Hey frens, 

Just reaching out to discuss best practices around the useContractWrite modes.
I understand that using recklesslyUnprepared has UX pitfalls; however, I suppose that if we still have the choice between recklesslyUnprepared and prepared, there should still be a use for recklesslyUnprepared, but I don't find one. 

In all of my recent apps, I've used prepared 100% of the times.

So please, feel free to chime in and share thoughts and feedbacks :) 

Peace :)

**Comment by SaadNiaziDev** [comment]
Signer
``
const {data:signer} = useSigner({chain:xx})
``

Creating contract instance 
``
const contract = useContract({
 abi: xxxxx , address: xxxxx ,signerOrProvider:signer });
``

Calling Write function in traditional way
`const harvest = async () => { 
		try {
			let tx = await contract.deposit( args  , {
                             from: address
			   } );
			await tx.wait();
			console.log( tx );
		} catch ( error ) {
			console.log( error )
		}
	}`


Not the Best approch but its simple to use for me
